ambry-api/src/main/java/com.github.ambry/config/ConnectionPoolConfig.java

```diff
@@ -60,6 +60,14 @@
   @Default("2")
   public final int connectionPoolMaxConnectionsPerPortSSL;
 
+  /**
+   * The maximum consecutive connect failures allowed before we clean up the available pool
+   */
+  @Config("connectionpool.max.connect.failures.to.clean.up.available.pool")
```

Comment:
Are we seeing any real impact of this change? If we do, we should tweak the failure detection configs and not change this.